### PR TITLE
feat(appointments): Disable the booking button while loading

### PR DESCRIPTION
**Pull request by pjrobertson**
The server can take a bit of time to response after clicking the 'Book the appointment button' (see below video - might be my server/email server is slow, hence why it taskes 8+ seconds!). To improve UX and to avoid the user clicking the button multiple times, we should disable the button while loading and/or add a loading icon.

https://github.com/nextcloud/calendar/assets/150431/ceacf31a-7ed6-47d9-8288-435a86d8f53c

Ideally, the code would also either replace the 'Book the appointment' text with a loading spinner, or grey out the entire popup window and add a loading icon, but that's beyond my ability 😿

### CODE REVIEW

File: src/components/Appointments/AppointmentDetails.vue [hunk 1]
```diff
@@ -125,12 +126,15 @@ export default {
...
 			this.$emit('save', {
 				slot: this.timeSlot,
 				description: this.description,
 				email: this.email,
 				displayName: this.displayName,
 				timeZone: this.timeZone,
+			}).then(() => {
+				this.isLoading = false
 			})
 		},
```
Comment (tcitworld) [inline src/components/Appointments/AppointmentDetails.vue:139]: **suggestion (non-blocking):** Another form is
```suggestion
		async save() {
			this.isLoading = true
			await this.$emit('save', {
				slot: this.timeSlot,
				description: this.description,
				email: this.email,
				displayName: this.displayName,
				timeZone: this.timeZone,
			})
			this.isLoading = false
		},
```

Comment (pjrobertson) [reply src/components/Appointments/AppointmentDetails.vue:139]: Nice, that looks simpler, more DRY. Happy to to adjust if you think that's best!

Comment (ChristophWurst) [reply src/components/Appointments/AppointmentDetails.vue:139]: yes please

Comment (pjrobertson) [reply src/components/Appointments/AppointmentDetails.vue]: Updated, thanks! :)

**Comment by pjrobertson** [comment]
> Let's also disable the inputs and textarea above while loading

Done :)

I also disabled closing of the modal whilst the submit form is loading.

Finally, I would really like to display a loading spinner in the button - I've added an attempt at that [here](https://github.com/nextcloud/calendar/pull/5244/commits/6b7ac63a14d6bade9e4b6bd3e0650cfae8a2ec0f), but I would appreciate feedback.

**Comment by pjrobertson** [comment]
Here's what the spinner looks like:


https://github.com/nextcloud/calendar/assets/150431/bbdb22c2-3529-49e0-ab3d-e63c26b260d1

**Comment by pjrobertson** [comment]
@tcitworld @ChristophWurst anything further I can do on this PR to get it merged? Thanks!

File: src/components/Appointments/AppointmentDetails.vue [hunk 2]
```diff
@@ -124,14 +131,18 @@ export default {
...
+			this.isLoading = true
+			await this.$emit('save', {
 				slot: this.timeSlot,
 				description: this.description,
 				email: this.email,
 				displayName: this.displayName,
 				timeZone: this.timeZone,
 			})
+			this.canClose = true
+			this.isLoading = false
```
Comment (ChristophWurst) [inline src/components/Appointments/AppointmentDetails.vue:145]: tip: wrap in a try-finally where you reset the two flags in the finally block. that ensures the state is also reset for errors.

**Review by ChristophWurst — COMMENTED** [review]
Code looks good

File: src/components/Appointments/AppointmentDetails.vue (hunk 2 above)
Comment (pjrobertson) [reply src/components/Appointments/AppointmentDetails.vue:145]: Nice tip, thanks! I have updated the code 👍

**Comment by pjrobertson** [comment]
@ChristophWurst  - sorry to keep bugging you on this. Who should I request to review the PR (that's blocking the merge)?

Thanks!

**Review by ChristophWurst — APPROVED** [review]

**Comment by pjrobertson** [comment]
🎉 Thanks Christoph! I hope to try and work on a few more bugs soon as well :-)

**Comment by ChristophWurst** [comment]
Awesome!

**Pull request merged**